AEM/optimum-habana-perf.md

```diff
@@ -1,48 +1,57 @@
+cd Gaudi-tutorials/PyTorch/Hugging_Face_pipelines/Benchmarking_on_Optimum-habana_with_fp8
 ```
-{:.note}**NOTE:** The Huggingface model file size might be large, so it is recommended to use an external disk as the Huggingface hub folder. Export the HF\_HOME environment variable to the external disk and then export the mount point into the Docker instance. ex: "-e HF\_HOME=/mnt/huggingface -v /mnt:/mnt"
-
-### Install required packages inside docker
-First, install the optimum-habana:
+### Docker Build
+To build the image from the Dockerfile, please follow below command to build the optimum-habana-text-gen image.
 ```bash
-pip install --upgrade-strategy eager optimum[habana]
+docker build --no-cache -t optimum-habana-text-gen:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f Dockerfile .
 ```
-
-Second, install the requirements:
+### Docker Run
+After docker build, users could follow below command to run and docker instance and users will be in the docker instance under text-generation folder.
 ```bash
-pip install -r requirements.txt
+docker run -it --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none   --cap-add=ALL --privileged=true  --net=host --ipc=host optimum-habana-text-gen:latest
 ```
+> [!NOTE]
+> The Huggingface model file size might be large, so we recommend to use an external disk as Huggingface hub folder. \
+> Please export HF_HOME environment variable to your external disk and then export the mount point into docker instance. \
+> ex: "-e HF_HOME=/mnt/huggingface -v /mnt:/mnt"
+
+# Run Benchmark with Benchmark.py
+Benchmark script will run all the models with different input len, output len and batch size and generate a report to compare all published numbers in [Gaudi Model Performance](https://www.intel.com/content/www/us/en/developer/platform/gaudi/model-performance.html).  
 
-For `run_lm_eval.py`:
+## Gaudi3
+Different json file are provided for different Gaudi Software version like 1.19 and 1.20 on Gaudi3.
+To do benchmarking on a machine with 8 Gaudi3 cards, just run the below command inside the docker instance. 
 ```bash
-pip install -r requirements_lm_eval.txt
+python3 Benchmark.py
 ```
-
-Then, to use [DeepSpeed-inference](https://docs.habana.ai/en/latest/PyTorch/DeepSpeed/Inference_Using_DeepSpeed.html), install DeepSpeed as follows: 
+## Gaudi2
+To do benchmarking on a machine with 8 Gaudi2 cards, just run the below command instead inside the docker instance. 
 ```bash
-pip install git+https://github.com/HabanaAI/DeepSpeed.git@1.20.0
+GAUDI_VER=2 python3 Benchmark.py
 ```
-
-
-# Tensor quantization statisics measurement
+## HTML Report
+A html report will be generated under a folder with timestamp.  
+The [html report](https://github.com/HabanaAI/Gaudi-tutorials/tree/main/PyTorch/Hugging_Face_pipelines/Benchmarking_on_Optimum-habana_with_fp8#html-report) also has a perf_ratio column to compare the measure numbers with reference perf numbers. 
```

Comment:
What are "measure numbers"? Are these previously generated numbers?

Reply:
I called previously generated numbers as reference perf numbers here, so measured numbers are from this benchmark run.  let me know any modification needed.

Reply:
Let's use "previous benchmark results".

Reply:
addressed it accordingly.